### PR TITLE
Add more granular OTEL buckets

### PR DESCRIPTION
**Pull request by jjti**
We're going to be using histogram-based samples for many of the dashboards of this feature. Many of those graphs will use percentiles. We're going to be calculating those percentiles via `histogram_quantile`: https://prometheus.io/docs/prometheus/latest/querying/functions/#histogram_quantile

But that's a bit error prone, the amount of error being related to the bucket widths. This adds more buckets so that percentile calculation from these metrics has less error.

https://prometheus.io/docs/practices/histograms/

Quantile error (see below for details) | Error is limited in the dimension of observed values by the width of the relevant bucket. | Error is limited in the dimension of φ by a configurable value.
-- | -- | --

### CODE REVIEW

File: agent/hcp/telemetry/otel_sink.go [hunk 1]
```diff
@@ -118,6 +118,8 @@ func NewOTELSink(opts *OTELSinkOpts) (*OTELSink, error) {
 	}, nil
 }
 
+var _ gometrics.MetricSink = &OTELSink{}
```
Comment (Achooo) [inline agent/hcp/telemetry/otel_sink.go:121]: Why was this added?

Comment (jjtimmons) [reply agent/hcp/telemetry/otel_sink.go:121]: This was just me trying to figure out which interface this type maps to and implements. Partly documentation and partly (generally) checking that an implementation conforms to an expected interface

**Review by Achooo — APPROVED** [review]

**Comment by Achooo** [comment]
Just needs some test updates for the metric data to pass CI
